src/test/java/gradle/cucumber/EchoSteps.java

```diff
@@ -4,74 +4,77 @@
 import cucumber.api.java.en.Given;
 import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
-import org.apache.http.HttpEntity;
-import org.apache.http.client.ResponseHandler;
 import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpGet;
 import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
 import org.apache.http.entity.StringEntity;
 import org.apache.http.impl.client.BasicResponseHandler;
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.impl.client.HttpClients;
-import org.apache.http.util.EntityUtils;
 
 import java.io.IOException;
 import java.net.*;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-public class echoStepDefs {
+public class EchoSteps {
   private CloseableHttpClient httpclient;
   private CloseableHttpResponse response;
   private HttpGet httpGet;
-  private String responseBody;
   private HttpPost httpPost;
+  private static int DEFAULT_PORT = 5000;
+  private static String HOST = "127.0.0.1";
 
-  @Given("the server is running")
-  public void serverIsRunning() throws IOException {
+  private static boolean serverIsRunning(int port) throws IOException {
     boolean result = false;
     try {
-      (new Socket("127.0.0.1", 3000)).close();
+      (new Socket("127.0.0.1", port)).close();
       result = true;
-    } catch(SocketException e) {
+    } catch (SocketException e) {
       System.out.println(e);
     }
-    assertTrue(result);
+    return result;
+  }
+
+  @Given("the server is running")
+  public void serverIsRunning() throws IOException {
+    assertTrue(serverIsRunning(DEFAULT_PORT));
   }
 
   @When("^I \"POST\" \"([^\"]*)\" to \"([^\"]*)\"$")
   public void iTo(String body, String path) throws Throwable {
     httpclient = HttpClients.createDefault();
-    httpPost = new HttpPost("http://localhost:3000" + path);
-    response = httpclient.execute(httpPost);
+    URI uri = new URIBuilder()
+      .setScheme("http")
+      .setHost(HOST)
+      .setPort(DEFAULT_PORT)
+      .setPath(path)
+      .build();
+
+    httpPost = new HttpPost(uri);
     httpPost.setEntity(new StringEntity(body));
-    CloseableHttpResponse response = httpclient.execute(httpPost);
-    try {
-      HttpEntity entity2 = response.getEntity();
-      EntityUtils.consume(entity2);
-    } finally {
-      response.close();
-    }
+    response = httpclient.execute(httpPost);
   }
 
   @When("^I request \"GET\" \"([^\"]*)\"$")
   public void iRequest(String path) throws Throwable {
     httpclient = HttpClients.createDefault();
-    httpGet = new HttpGet("http://localhost:3000" + path);
+    URI uri = new URIBuilder()
+      .setScheme("http")
+      .setHost(HOST)
+      .setPort(DEFAULT_PORT)
+      .setPath(path)
+      .build();
```

Comment:
Just FYI--this is a pattern you will see EVERYWHERE in Java. It would be a great pattern to understand and use to build classes in your server.

Reply:
https://dzone.com/articles/design-patterns-the-builder-pattern